scripts/irods/test/test_ichksum.py

```diff
@@ -378,3 +380,30 @@ def test_ichksum_groups_objects_appropriately_when_the_recursive_flag_is_set__is
            data_object_3)
         self.assertTrue(re.match(pattern, out))
 
+    def test_ichksum_honors_the_size_in_the_catalog_when_computing_checksums__issue_5401(self):
+        data_object = os.path.join(self.admin.session_collection, 'foo')
+        self.admin.assert_icommand(['istream', 'write', data_object], input='the data')
+
+        # Make the catalog report the wrong size of the replica on disk.
+        # This change will cause icommands such as iget and istream to return an empty string.
+        self.admin.assert_icommand(['iadmin', 'modrepl', 'logical_path', data_object, 'replica_number', '0', 'DATA_SIZE', '0'])
+        self.admin.assert_icommand(['istream', 'read', data_object])
+
+        # Show that the calculated checksum matches the SHA256 checksum of an empty string.
+        replica_checksum = '47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU='
+        self.admin.assert_icommand(['ichksum', '-f', data_object], 'STDOUT', ['sha2:' + replica_checksum])
+        self.assertEqual(replica_checksum, base64.b64encode(hashlib.sha256('').digest()))
+
```

Comment:
can we have an additional assertion with size 3 that checksum equals sha256('the')?